src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -208,6 +200,154 @@ public static void LoadSatelliteAssembly(byte[] dllBytes)
+                }
+                if (method == null)
+                {
+                    throw new InvalidOperationException(string.Format(SR.CannotResolveManagedEntrypoint, asyncName, mainAssemblyName));
+                }
+            }
+
+            if (waitForDebugger)
+            {
+                Interop.Runtime.SetEntryPointBreakpoint(method.MetadataToken);
+            }
+
+            object[] argsToPass = System.Array.Empty<object>();
+            Task<int>? result = null;
+            var parameterInfos = method.GetParameters();
+            if (parameterInfos.Length > 0 && parameterInfos[0].ParameterType == typeof(string[]))
+            {
+                argsToPass = new object[] { args ?? System.Array.Empty<string>() };
+            }
+            if (method.ReturnType == typeof(void))
+            {
+                method.Invoke(null, argsToPass);
+            }
+            else if (method.ReturnType == typeof(int))
+            {
+                int intResult = (int)method.Invoke(null, argsToPass)!;
+                result = Task.FromResult(intResult);
+            }
+            else if (method.ReturnType == typeof(Task))
+            {
+                Task methodResult = (Task)method.Invoke(null, argsToPass)!;
+                TaskCompletionSource<int> tcs = new TaskCompletionSource<int>();
+                result = tcs.Task;
+                methodResult.ContinueWith((t) =>
+                {
+                    if (t.IsFaulted)
+                    {
+                        tcs.SetException(t.Exception!);
+                    }
+                    else
+                    {
+                        tcs.SetResult(0);
+                    }
+                }, TaskScheduler.Default);
+            }
+            else if (method.ReturnType == typeof(Task<int>))
+            {
+                result = (Task<int>)method.Invoke(null, argsToPass)!;
+            }
+            else
+            {
+                throw new InvalidOperationException(SR.Format(SR.ReturnTypeNotSupportedForMain, method.ReturnType.FullName));
+            }
+            return result;
+        }
+
+        private static string GeneratedInitializerClassName = "System.Runtime.InteropServices.JavaScript.__GeneratedInitializer";
+        private static string GeneratedInitializerMethodName = "__Register_";
+
+        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Dynamic access from JavaScript")]
+        [UnconditionalSuppressMessage("Trimming", "IL2075", Justification = "Dynamic access from JavaScript")]
+        public static Task BindAssemblyExports(string? assemblyName)
+        {
+            if (string.IsNullOrEmpty(assemblyName))
+            {
+                throw new MissingMethodException("Missing assembly name");
+            }
+
+            Assembly assembly = Assembly.LoadFrom(assemblyName);
```

Comment:
What if user passes an not-existing assembly name?

Reply:
I just moved the logic from C/JS. Here it will have nice exception saying it can't find "/dummy.dll" file.